transitions/generate_patient_id_on_patients.js

```diff
@@ -0,0 +1,11 @@
+var transitionUtils = require('./utils');
+
+module.exports = {
+  filter: function(doc) {
+    return doc.type === 'person' &&
```

Comment:
We also need to keep in mind that `id` generation is typically limited because they are short by design because they are used in SMS.  So using them where if they are not used will be a waste.  Also other transitions will not complete (sync they all run synchronously in memory on a single change) until this one completes.  So it potentially slows down the transition processing chain.  I'm in favor of having an easy way in configuration to enable/disable which documents are assigned patient ids, similar to how registrations behave by having a boolean expression that is evaluated, and by default it should be off.  There is a cost for generating these types of IDs and we've seen instances in the past that are "stuck" because the entire pool of IDs are in use.